threading/smartptrs.nim

```diff
@@ -134,7 +132,7 @@ proc newSharedPtr*[T](t: typedesc[T]): SharedPtr[T] =
     result.val = cast[typeof(result.val)](allocShared0(sizeof(result.val[])))
   else:
     result.val = cast[typeof(result.val)](allocShared(sizeof(result.val[])))
-  int(result.val.counter) = 0
+  int(result.val.counter) = 1
```

Comment:
But previously this implied that `default(SharedPtr[T])` was useful, now it is not.

Reply:
Ah, great point! I wondered why you started at 0 previously. I'll push a fix to restore the 0 based count.

Reply:
```suggestion
  int(result.val.counter) = 0
```